src/lightning/pytorch/trainer/setup.py

```diff
@@ -184,8 +193,17 @@ def _log_device_info(trainer: "pl.Trainer") -> None:
     if TPUAccelerator.is_available() and not isinstance(trainer.accelerator, TPUAccelerator):
         rank_zero_warn("TPU available but not used. You can set it by doing `Trainer(accelerator='tpu')`.")
 
-    if _IPU_AVAILABLE and not isinstance(trainer.accelerator, IPUAccelerator):
-        rank_zero_warn("IPU available but not used. You can set it by doing `Trainer(accelerator='ipu')`.")
+    if _IPU_AVAILABLE:
+        if not _LIGHTNING_GRAPHCORE_AVAILABLE:
+            raise ModuleNotFoundError(
+                "You are running on IPU machine but you have not installed `lightning-graphcore`"
+                f" extension is  {str(_LIGHTNING_GRAPHCORE_AVAILABLE)}."
```

Comment:
There's on extra space here. Same for HPU below

```suggestion
                f" extension is {str(_LIGHTNING_GRAPHCORE_AVAILABLE)}."
```